File: MAP/lab2/src/models/IfStmt.java
package models;

import utils.IDictionary;
import utils.IExeStack;

public class IfStmt implements Statement {

    private IExpression exp;
    private Statement thenS;
    private Statement elseS;

    public IfStmt(IExpression _exp, Statement _thenS, Statement _elseS){
        exp = _exp;
        thenS = _thenS;
        elseS = _elseS;
    }

    @Override
    public PrgState execute(PrgState state) {
        IDictionary <String, Integer> dict = state.getDict();
        IExeStack<Statement> stack = state.getStack();
        int val = exp.eval(dict);

        if(val != 0)
            stack.push(thenS);
        else
            stack.push(elseS);

        return state;
    }

    @Override
    public String toString(){
        return "IF("+ exp + ") THEN(" +thenS + ")ELSE(" + elseS + ") ;";
    }
}

File: MAP/lab7/lab7/Models/IfStmt.cs
namespace lab7.Models
{
    public class IfStmt : IStatement
    {
        private IStatement thenS;
        private IStatement elseS;
        private IExpression exp;

        public IfStmt(IExpression exp, IStatement thenS, IStatement elseS)
        {
            this.thenS = thenS;
            this.elseS = elseS;
            this.exp = exp;
        }

        public PrgState execute(PrgState state)
        {
            if(exp.eval(state.Dict) != 0)
                state.Stack.push(thenS);
            else
                state.Stack.push(elseS);

            return state;

        }

        public override string ToString()
        {
            return "IF("+ exp + ") THEN(" +thenS + ")ELSE(" + elseS + ")";
        }
    }
}